### PR TITLE
Added spigot only updater.

### PR DESCRIPTION
**Pull request by Arcratist**

### CODE REVIEW

**Comment by DoctorMacc** [comment]
Seems fine to me, but it would be nice if we could roll out auto-updating to all platforms if possible.

**Comment by Arcratist** [comment]
Spigot has an update folder that replaces the plugin jar with its contents at startup. Other platforms will be much more complicated. Ill take a look at potential solutions tomorrow some time.

File: connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public UpdateCommand(GeyserConnector connector, String name, String description, String permission) {
+        super(name, description, permission);
+        this.connector = connector;
+    }
+
+    @Override
+    public void execute(CommandSender sender, String[] args) {
+        //Currently only Spigot works
+        String platform = GeyserConnector.getInstance().getPlatformType().getPlatformName();
+        if (!platform.equals("Spigot")) {
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java:59]: This should use the enums for `PlatformType` instead

File: connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                    if (buildXML.startsWith("<buildNumber>")) {
+                        int latestBuildNum = Integer.parseInt(buildXML.replaceAll("<(\\\\)?(/)?buildNumber>", "").trim());
+                        int buildNum = Integer.parseInt(gitProp.getProperty("git.build.number"));
+                        //Compare build numbers.
+                        if (latestBuildNum == buildNum) {
+                            sender.sendMessage(LanguageUtils.getLocaleStringLog("geyser.commands.version.no_updates"));
+                        } else {
+                            sender.sendMessage(LanguageUtils.getLocaleStringLog("geyser.commands.update.outdated", (latestBuildNum - buildNum)));
+
+                            String downloadUrl = "https://ci.nukkitx.com/job/GeyserMC/job/Geyser/job/" + URLEncoder.encode(gitProp.getProperty("git.branch"), StandardCharsets.UTF_8.toString()) + "/lastSuccessfulBuild/artifact/bootstrap/" + platform.toLowerCase() + "/target/Geyser" + (platform.equals("Standalone") ? ".jar" : "-" + platform + ".jar");
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java:84]: Same as above

File: connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                            //TODO: Replace with regex. Gotta figure that one out though XD
+                            int divIndex = checksumData.indexOf("<div class=\"md5sum\"");
+                            divIndex = checksumData.indexOf(">", divIndex);
+                            int endDivIndex = checksumData.indexOf("</div>", divIndex);
+                            String remoteChecksum = checksumData.substring(divIndex + 1, endDivIndex).replace("MD5: ", "").trim();
+
+                            File updateFolder = GeyserConnector.getInstance().getBootstrap().getConfigFolder().getParent().resolve("update").toFile();
+                            updateFolder.mkdir();
+
+                            File downloadedFile = new File(updateFolder, platform.equals("Standalone") ? "Geyser.jar" : "Geyser-" + platform + ".jar");
```
Comment (rtm516) [inline connector/src/main/java/org/geysermc/connector/command/defaults/UpdateCommand.java:96]: Same as above

**Review by rtm516 — CHANGES_REQUESTED** [review]
Comments should also have a space after the `//`
EG: `// Currently only Spigot works`

**Comment by Arcratist** [comment]
Made a few changes

**Review by rtm516 — APPROVED** [review]

**Review by lukeeey — CHANGES_REQUESTED** [review]
Please use this URL as it contains the hashes as well
```
https://ci.nukkitx.com/job/GeyserMC/job/Geyser/job/master/lastSuccessfulBuild/api/xml?depth=2&tree=fingerprint[fileName,hash]{0,},artifacts[fileName,relativePath]
```
Also, if possible, feel free to use an XML parser (even the built-in one) to parse the above

**Comment by YHDiamond** [comment]
I think this should be updated to the new CI URL.

**Comment by YHDiamond** [comment]
There is a plugin that @AlysaaKobe made that does this too just saying. github.com/YHDiamond/GeyserUpdater

**Comment by Camotoy** [comment]
We're going to close this PR, at least for the time being. Some form of updater is not off the table, but it will be for all platforms, not just Spigot. For now, there are community methods of autoupdating that can be used.

Thank you for the PR anyway!

**Pull request closed without merging**